File: Baekjoon_Online_Judge/NO_01032/1032.c
#include <stdio.h>
/*
 -------------------------------------- 
 | Created by happiness96             | 
 | Year 2020                          | 
 | Month 09                           | 
 | Day 02                             | 
 | 명령 프롬프트                       | 
 -------------------------------------- 
*/

int main(){
    int N;
    char form[50];

    scanf("%d", &N);
    scanf("%s", &form);

    for (int i = 0; i < N - 1; i ++){
        char sub_s[50];

        scanf("%s", &sub_s);

        for (int ind = 0; sub_s[ind] != '\0'; ind ++){
            if (sub_s[ind] != form[ind]) form[ind] = '?';
        }
    }

    printf("%s", form);

    return 0;
}

File: Baekjoon_Online_Judge/NO_01032/1032.py
# -*- encoding: utf-8 -*-
import sys
r_input = sys.stdin.readline

# -------------------------------------- #
# | Created by happiness96             | #
# | Year 2020                          | #
# | Month 09                           | #
# | Day 02                             | #
# | 명령 프롬프트                       | #
# -------------------------------------- #


if __name__ == "__main__":
    N = int(r_input())

    form = list(r_input().rstrip())

    for _ in range(N - 1):
        file_name = r_input().rstrip()

        for ind, val in enumerate(file_name):
            if val != form[ind]:
                form[ind] = '?'

    print(''.join(form))
